pyiron/vasp/base.py

```diff
@@ -727,58 +727,63 @@ def write_magmoms(self):
         Write the magnetic moments in INCAR from that assigned to the species
         """
         if any(self.structure.get_initial_magnetic_moments().flatten()):
-            final_cmd = "   ".join(
-                [
-                    " ".join([str(spinmom) for spinmom in spin])
-                    if isinstance(spin, list) or isinstance(spin, np.ndarray)
-                    else str(spin)
-                    for spin in self.structure.get_initial_magnetic_moments()[
-                        self.sorted_indices
-                    ]
-                ]
-            )
-            s.logger.debug("Magnetic Moments are: {0}".format(final_cmd))
-            if "MAGMOM" not in self.input.incar._dataset["Parameter"]:
-                self.input.incar["MAGMOM"] = final_cmd
             if "ISPIN" not in self.input.incar._dataset["Parameter"]:
                 self.input.incar["ISPIN"] = 2
-            if any(
-                [
-                    True
-                    if isinstance(spin, list) or isinstance(spin, np.ndarray)
-                    else False
-                    for spin in self.structure.get_initial_magnetic_moments()
-                ]
-            ):
-                self.input.incar["LNONCOLLINEAR"] = True
-                if (
-                    self.spin_constraints
-                    and "M_CONSTR" not in self.input.incar._dataset["Parameter"]
-                ):
-                    self.input.incar["M_CONSTR"] = final_cmd
-                if (
-                    self.spin_constraints
-                    or "M_CONSTR" in self.input.incar._dataset["Parameter"]
-                ):
-                    if "ISYM" not in self.input.incar._dataset["Parameter"]:
-                        self.input.incar["ISYM"] = 0
-                if (
-                    self.spin_constraints
-                    and "LAMBDA" not in self.input.incar._dataset["Parameter"]
-                ):
-                    raise ValueError(
-                        "LAMBDA is not specified but it is necessary for non collinear calculations."
-                    )
-                if (
-                    self.spin_constraints
-                    and "RWIGS" not in self.input.incar._dataset["Parameter"]
+            if self.input.incar["ISPIN"] != 1:
+                final_cmd = "   ".join(
+                    [
+                        " ".join([str(spinmom) for spinmom in spin])
+                        if isinstance(spin, list) or isinstance(spin, np.ndarray)
```

Comment:
Better to write `if  isinstance(spin, (list, np.ndarray))`

Reply:
, I thought, we have tests for this? The github highlighting is a little bit strange as I only added an outer if statement 😄

Reply:
done

Reply:
> 
> 
> , I thought, we have tests for this? The github highlighting is a little bit strange as I only added an outer if statement 😄

Codeacy only highlights pep8 violations. It does not suggest better coding styles unfortunately!